pkg/daemon/ceph/osd/kms/kms.go

```diff
@@ -181,6 +181,20 @@ func ValidateConnectionDetails(clusterdContext *clusterd.Context, clusterSpec *c
 		}
 	}
 
+	// Validate potential Ceph-CSI config map presence
+	if clusterSpec.Security.KeyManagementService.IsCSIEnabled() {
+		csiEncryptionConfig, err := clusterdContext.Clientset.CoreV1().ConfigMaps(ns).Get(ctx, clusterSpec.Security.KeyManagementService.CSIConnectionDetailsCMName, metav1.GetOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "failed to fetch kms csi config map %q", clusterSpec.Security.KeyManagementService.CSIConnectionDetailsCMName)
+		}
+
+		// Check for empty config
+		config, ok := csiEncryptionConfig.Data[kmsCSIConfigMapNameKey]
+		if !ok || len(config) == 0 {
```

Comment:
Users can create an empty config and create a ceph cluster. May be he can update KMS details in configmap later also. is it an option available?

Reply:
Today no, but this can change if we want to support such a scenario, I just wonder what's the point of creating something empty?

Reply:
Yes, correct, was checking different possible options.